.github/workflows/dirty-waters.yml

```diff
@@ -0,0 +1,49 @@
+name: Dirty Waters
+on:
+  pull_request: ~
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 3 * * *"
+  workflow_dispatch: ~
+
+permissions: read-all
+
+jobs:
+  analyze:
+    name: Analysis
+    runs-on: ubuntu-24.04
+    permissions:
+      pull-requests: write # To comment on a Pull Request
+    steps:
+      - name: Harden runner
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.2
```

Comment:
This version **MUST** be synced with the rest of the project before this is merged, see https://github.com/search?q=repo%3Aericcornelissen%2Fshescape%20step-security%2Fharden-runner&type=code